Do not process lazy objects

Kronuz opened the pull request:
Summary: This prevents `_find_torch_objects()` from resolving lazy objects in the modules. This only can happen when Using Cinder and Lazy Imports.

Test Plan: Run all existing tests

Differential Revision: D46542598



cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @ipiszy @msaroufim

ezyang reviewed (changes requested):
This seems unlikely to be right. The semantics of `find_torch_objects` requires eager imports, because we are walking the module hierarchy looking to find identifiers. If you load modules lazily (and don't force them to be loaded here), then we will fail to populate the allowed functions list entirely.

CI coverage is unlikely to be sufficient because none of the configs are running Cinder.

jansel reviewed (changes requested):
I agree with ezyang, this seems incorrect.

Kronuz committed:
Do not process lazy objects (#103617)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/103617

This prevents `_find_torch_objects()` from resolving lazy objects in the modules. This only can happen when Using Cinder and Lazy Imports.

Test Plan:
Run all existing tests

 ---

1. Enable lazy imports (`enable_lazy_imports = True`) for `run_cli`, in `TARGETS`.

2. Run the tool (full output at P771163039):

```
CLI_DISABLE_BEST_PRACTICES_ENFORCEMENT=1 buck2 run fblearner/flow/projects/fluent2:run_cli -- validate_domain demo.sparse_multi_class.domain --mode=canary_flow_remote --secure_group=oncall_instagram_cinder --oncall=cinder --entitlement=integration_tests --disable-memoization 06-14-2023

...

Validation successful: demo.sparse_multi_class.domain
```

Differential Revision: D46542598

fbshipit-source-id: 766595c1dd2f1c7fed953800985420bf62caa992
